File: Simon_C#/Simon_C_Sharp/LabelEspecial.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Drawing;

namespace Simon_C_Sharp
{
    public class LabelEspecial : Label
    {
        private Color _colorEnfocado;
        private Color _colorDesenfocado;

        public LabelEspecial()
        {
            this.MouseDown += new MouseEventHandler(Presionar);
            this.MouseUp += new MouseEventHandler(Soltar);
        }

        public Color ColorEnfocado
        {
            get { return _colorEnfocado; }
            set { _colorEnfocado = value; }
        }

        public Color ColorDesenfocado
        {
            get { return _colorDesenfocado; }
            set { _colorDesenfocado = value; }
        }

        public void Presionar(Object sender, EventArgs e)
        {
            this.BackColor = this._colorEnfocado;
        }

        public void Soltar(Object sender, EventArgs e)
        {
            this.BackColor = this._colorDesenfocado;
        }


    }
}

File: Simon_Visual_Basic/PruebaLabels 18/Backup/PruebaLabels/LabelEspecial.vb
Option Strict On

Public Class LabelEspecial
    Inherits Label


    Private _colorEnfocado As Color
    Private _colorDesenfocado As Color

    Public Property ColorEnfocado() As Color
        Get
            Return Me._colorEnfocado
        End Get
        Set(ByVal value As Color)
            Me._colorEnfocado = value
        End Set
    End Property


    Public Property ColorDesenfocado() As Color
        Get
            Return Me._colorDesenfocado
        End Get
        Set(ByVal value As Color)
            Me._colorDesenfocado = value
        End Set
    End Property


    Public Sub Presionar(ByVal sender As Object, ByVal e As EventArgs) Handles Me.MouseDown
        Me.BackColor = Me._colorEnfocado
    End Sub

    Public Sub Soltar(ByVal sender As Object, ByVal e As EventArgs) Handles Me.MouseUp
        Me.BackColor = Me._colorDesenfocado
    End Sub








End Class
